hackathon_site/dashboard/frontend/src/pages/IncidentForm/IncidentForm.tsx

```diff
@@ -0,0 +1,419 @@
+import React from "react";
+import styles from "./IncidentForm.module.scss";
+import {
+    Typography,
+    Card,
+    CardContent,
+    RadioGroup,
+    FormControlLabel,
+    Radio,
+    Select,
+    MenuItem,
+    Button,
+    TextField,
+    FormControl,
+    InputLabel,
+    FormHelperText,
+    makeStyles,
+} from "@material-ui/core";
+import { useHistory, useLocation } from "react-router-dom";
+import { Formik, Field, FormikValues } from "formik";
+import * as Yup from "yup";
+import Header from "components/general/Header/Header";
+import ArrowLeft from "../../assets/images/icons/arrow-left-solid.svg";
+import { displaySnackbar } from "slices/ui/uiSlice";
+import { useDispatch } from "react-redux";
+
+export const INCIDENT_ERROR_MSG = {
+    state: "Please select an option",
+    qty: "Please indicate the quantity",
+    what: "Please indicate what happened to the hardware",
+    when: "Please indicate when this occurred",
+    where: "Please indicate where this occurred",
+};
+
+const useStyles = makeStyles({
+    typographyLabel: {
+        paddingBottom: "10px", // Adjust the value as per your requirement
+    },
+});
+
+const capitalizeFirstLetter = (str: string) => {
+    return str.charAt(0).toUpperCase() + str.slice(1);
+};
+
+const IncidentForm = () => {
+    const muiClasses = useStyles();
+
+    const dispatch = useDispatch();
+    let history = useHistory();
+    let location = useLocation();
+
+    // get info from url
+    const searchParams = new URLSearchParams(location.search);
+    let hardwareQuantity: number; // quantity used for dropdown
+    try {
+        const data = searchParams.get("data") ?? "";
+        const checkedOutOrder = JSON.parse(data); // parse string from url into an object
+        hardwareQuantity = checkedOutOrder?.quantityGranted; // set the hardware qty for dropdown
+    } catch {
+        hardwareQuantity = 0; // set the qty to 0 if there is no url
+    }
+
+    const validationSchema = Yup.object({
+        state: Yup.string().required(INCIDENT_ERROR_MSG.state),
+        qty: Yup.string().required(INCIDENT_ERROR_MSG.qty),
+        what: Yup.string().required(INCIDENT_ERROR_MSG.what),
+        when: Yup.string().required(INCIDENT_ERROR_MSG.when),
+        where: Yup.string().required(INCIDENT_ERROR_MSG.where),
+    });
+
+    const initialValues = {
+        state: "",
+        qty: "",
+        what: "",
+        when: "",
+        where: "",
+    };
+
+    const handleSubmit = async (values: FormikValues, { resetForm }: any) => {
+        // TODO: submit the form
+
+        resetForm();
+        // display success snackbar
+        dispatch(
+            displaySnackbar({
+                message: `Successfully submitted incident form!`,
+                options: {
+                    variant: "success",
+                },
+            })
+        );
+        // navigate back to previous page
+        history.goBack();
+    };
+
+    const createQuantityList = (number: number) => {
```

Comment:
This can be refactored. Both if/else cases do the same thing.

Reply:
What is supposed to happen if  `hardwareQuantity` is 0?

Reply:
@KarandeepLubana thanks for the feedback!

i kind of see what you mean with the if/else cases... but one of them only displays 0 while the other displays the quantity from 1 to the requested quantity.
just thought that the 0 shouldn't be displayed because they would have some quantity if they're reporting an incident.

hardwareQuantity would never be 0, the 0 state is just used when someone accidentally navigates to the page directly without clicking the incident form link in the table. i just added another commit to prevent submission in case that occurs!

hope that clarifies things!

Reply:
This is a common scenario where you don’t want users to have access to this route until they’ve added an order. You want to prevent users from directly accessing the route by entering the URL. 

Conditional redirects can be used to redirect users to a different page or display an error message when they try to access routes that should only be reached through specific actions or links. I suggest you implement a solution that uses conditional redirects.

Useful Links:
https://www.toptal.com/react/react-router-tutorial 
https://stackabuse.com/redirects-in-react-router/